caikit/core/model_management/local_model_trainer.py

```diff
@@ -132,29 +132,30 @@ def completion_time(self) -> Optional[datetime]:
 
         ## Interface ##
 
-        def get_status(self) -> TrainingStatus:
+        def get_info(self) -> TrainingInfo:
             """Every model future must be able to poll the status of the
             training job
             """
+
             # The worker was canceled while doing work. It may still be in the
             # process of terminating and thus still alive.
             if self._worker.canceled:
-                return TrainingStatus.CANCELED
+                return TrainingInfo(status=TrainingStatus.CANCELED)
 
             # If the worker is currently alive it's doing work
             if self._worker.is_alive():
-                return TrainingStatus.RUNNING
+                return TrainingInfo(status=TrainingStatus.RUNNING)
 
             # The worker threw outside of a cancellation process
             if self._worker.threw:
-                return TrainingStatus.ERRORED
+                return TrainingInfo(status=TrainingStatus.ERRORED)
```

Comment:
I think we should be able to get the error itself here?

Reply:
Yep. Will get that added and corresponding tests